discovery-provider/src/utils/prometheus_metric.py

```diff
@@ -73,5 +76,8 @@ def register_collector(cls, name, collector_func):
 
     @classmethod
     def populate_collectors(cls):
-        for collector in cls.registered_collectors.values():
-            collector()
+        for name, collector in enumerate(cls.registered_collectors):
+            try:
```

Comment:
@SidSethi however, we should future proof the node to skip faulty collectors and continue exposing the rest of our metrics.

Reply:
understood